### PR TITLE
Better starter docs

### PR DESCRIPTION
**Pull request by wuweiweiwu**
closes: #5866 

clarify what starters are :)

### CODE REVIEW

**Comment by wuweiweiwu** [comment]
Failed ci:

```
/home/travis/build/gatsbyjs/gatsby/packages/gatsby-link/src/index.js
  54:1  error  componentWillReceiveProps is deprecated since React 16.3.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops  react/no-deprecated
/home/travis/build/gatsbyjs/gatsby/packages/gatsby-react-router-scroll/src/ScrollContainer.js
  19:1  error  componentWillReceiveProps is deprecated since React 16.3.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops  react/no-deprecated
```

seems that some components in v1 still use deprecated lifecycles :(

File: docs/docs/gatsby-starters.md [hunk 1]
```diff
@@ -2,8 +2,9 @@
 title: 'Gatsby Starters'
 ---
 
-The Gatsby CLI tool lets you install "starters". These are partially built sites
-preconfigured to help you get moving faster on creating a certain type of site.
+The Gatsby CLI tool lets you install “starters”. These are boilerplate Gatsby sites maintained by the community.
+`gatsby new` helps you start your project by cloning the boilerplate, installing dependencies, and clearing git history.
+If you are familiar with `git` this is the same as `git clone <starter repo url> && rm -rf .git/ && yarn` (or `npm i`).
```
Comment (nihgwu) [inline docs/docs/gatsby-starters.md]: I'm wondering if it's necessary to explain about the `git` part? also you missed `cd folder`

Comment (wuweiweiwu) [reply docs/docs/gatsby-starters.md]: Gotcha. I think it makes sense to get rid of the parts starting from "If you are familar ..."

**Comment by m-allanson** [comment]
I've cherry-picked this across to the `master` branch so it'll show up on next.gatsbyjs.org too.

**Comment by wuweiweiwu** [comment]
Awesome!

**Pull request merged**